plugins/modules/api_modify.py

```diff
@@ -539,6 +540,22 @@ def sync_list(module, api, path, path_info):
                         index=index + 1,
                     )
                 )
+        for require_list in require_one_of:
+            found_req_keys = [rk for rk in require_list if rk in entry]
+            if len(found_req_keys) > 1:
```

Comment:
I don't think this check should be done here. `required_one_of` should mean "at least one of them must be there". If you also want to make sure that some fields are mutually exclusive, better add another list of lists `mutually_exclusive`.

Reply:
Added mutually_exclusive as suggested.